Misc/NEWS.d/next/Library/2024-12-11-17-44-36.gh-issue-127794.VwmRsp.rst

```diff
@@ -0,0 +1,2 @@
+The :meth:`email.message.Message.add_header` method now validates header
+field names according to :rfc:`5322`.
```

Comment:
This needs rewriting now, too.  Something like ":meth:email.message.Message now validates header field names according to :rfc`5322` when headers are added via `add_header` or `__setitem__` and raises a ValueError if they contain invalid characters."